code/game/objects/items/paintkit.dm

```diff
@@ -18,95 +21,156 @@
 		user.drop_item()
 		qdel(src)
 
+/obj/item/device/kit/proc/can_customize(var/obj/item/I)
+	return is_type_in_list(I, allowed_types)
+
+/obj/item/device/kit/proc/set_info(var/kit_name, var/kit_desc, var/kit_icon, var/kit_icon_file = CUSTOM_ITEM_OBJ, var/kit_icon_override_file = CUSTOM_ITEM_MOB, var/additional_data)
+	new_name = kit_name
+	new_desc = kit_desc
+	new_icon = kit_icon
+	new_icon_file = kit_icon_file
+	new_icon_override_file = kit_icon_override_file
+
+	for(var/path in splittext(additional_data, ", "))
+		allowed_types |= text2path(path)
+
+/obj/item/device/kit/proc/customize(var/obj/item/I, var/mob/user)
+	if(can_customize(I))
+		I.name = new_name ? new_name : I.name
+		I.desc = new_desc ? new_desc : I.desc
+		I.icon = new_icon_file ? new_icon_file : I.icon
+		I.icon_override = new_icon_override_file ? new_icon_override_file : I.icon_override
+		if(new_icon)
+			I.icon_state = new_icon
+		var/obj/item/clothing/under/U = I
+		if(istype(U))
+			U.worn_state = I.icon_state
+			U.update_rolldown_status()
+		use(1, user)
+
+// Generic use
+/obj/item/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/device/kit))
+		var/obj/item/device/kit/K = W
+		K.customize(src, user)
+		return
+
+	..()
+
 // Root hardsuit kit defines.
 // Icons for modified hardsuits need to be in the proper .dmis because suit cyclers may cock them up.
 /obj/item/device/kit/suit

```

Comment:
Can you add a subtype or a provision for `/obj/item/clothing/suit/storage/hooded`? Because those store the helmet item internally, so you can't really apply the kit to the hood itself to change it